test/integration_test/action_test.go

```diff
@@ -21,67 +20,63 @@ import (
 )
 
 func TestAction(t *testing.T) {
-	// Create secrets on source and destination
-	// Since the secrets need to be created on the destination before migration
-	// is triggered using the API instead of spec factory in torpedo
-	secret := createSecret()
-	setSecret(t, secret, "src")
-	setSecret(t, secret, "dest")
+
+	setupOnce(t)
 
 	t.Run("actionFailoverTest", actionFailoverTest)
 }
 
-func createSecret() *v1.Secret {
-	return &v1.Secret{
-		ObjectMeta: meta_v1.ObjectMeta{
-			Name:      "volume-secrets",
-			Namespace: "kube-system",
-		},
-		StringData: map[string]string{
-			"mysql-secret": "supersecretpassphrase",
-		},
+func setupOnce(t *testing.T) {
+	funcCreateSecret := func() {
+		_ = createSecret(
+			t,
+			"volume-secrets",
+			map[string]string{
+				"mysql-secret": "supersecretpassphrase",
+			})
 	}
+	funcCreateSecret()
+	executeOnDestination(t, funcCreateSecret)
 }
 
-func setSecret(t *testing.T, secret *v1.Secret, cluster string) {
-	if cluster == "dest" {
-		logrus.Infof("creating secret in destination")
-		err := setDestinationKubeConfig()
-		require.NoError(t, err, "failed to set kubeconfig to destination cluster: %v", err)
-
-		defer func() {
-			err := setSourceKubeConfig()
-			require.NoError(t, err, "failed to set kubeconfig to source cluster: %v", err)
-		}()
-	} else {
-		logrus.Infof("creating secret in source")
+func createSecret(t *testing.T, secret_name string, secret_map map[string]string) *v1.Secret {
+	secret := &v1.Secret{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name:      secret_name,
+			Namespace: "kube-system",
+		},
+		StringData: secret_map,
 	}
-
-	_, err := core.Instance().CreateSecret(secret)
+	secretObj, err := core.Instance().CreateSecret(secret)
 	if !errors.IsAlreadyExists(err) {
 		require.NoError(t, err, "failed to create secret for volumes")
 	}
+	return secretObj
 }
 
-func deleteNamespace(t *testing.T, namespace string, cluster string) {
-	if cluster == "dest" {
-		logrus.Infof("deleting namespace %v in destination", namespace)
-		err := setDestinationKubeConfig()
-		require.NoError(t, err, "failed to set kubeconfig to destination cluster: %v", err)
-
-		defer func() {
-			err := setSourceKubeConfig()
-			require.NoError(t, err, "failed to set kubeconfig to source cluster: %v", err)
-		}()
-	} else {
-		logrus.Infof("deleting namespace %v in source", namespace)
+func cleanup(t *testing.T, namespace string) {
+	funcDeleteNamespace := func() {
+		err := core.Instance().DeleteNamespace(namespace)
+		if err != nil {
+			logrus.Infof("Error deleting namespace %s: %v\n", namespace, err)
+		}
 	}
+	funcDeleteNamespace()
+	executeOnDestination(t, funcDeleteNamespace)
```

Comment:
this is a great idea, as an improvement we can also have something like `executeOnSource(...)`